app/src/main/java/net/squanchy/schedule/ScheduleService.java

```diff
@@ -32,53 +41,83 @@
     public Observable<Schedule> schedule() {
         Observable<FirebaseSchedule> sessionsObservable = dbService.sessions();
         Observable<FirebaseSpeakers> speakersObservable = dbService.speakers();
+        final Observable<FirebaseDays> daysObservable = dbService.days();
 
-        return Observable.combineLatest(
-                sessionsObservable,
-                speakersObservable,
-                combineIntoSchedule()
-        ).subscribeOn(Schedulers.io());
+        return Observable.combineLatest(sessionsObservable, speakersObservable, combineSessionsAndSpeakers())
+                .map(mapEventsToDays())
+                .withLatestFrom(daysObservable, combineDaysToGetASchedule())
+                .subscribeOn(Schedulers.io());
     }
 
-    private BiFunction<FirebaseSchedule, FirebaseSpeakers, Schedule> combineIntoSchedule() {
-        return (apiSchedule, apiSpeakers) -> {
-            List<SchedulePage> pages = map(apiSchedule.days, toSchedulePage(apiSchedule, apiSpeakers));
+    @NonNull
+    private BiFunction<HashMap<Integer, List<Event>>, FirebaseDays, Schedule> combineDaysToGetASchedule() {
```

Comment:
You need to sort methods top-down in this class, it's tricky to follow the code otherwise.

Nit: `combineDaysIntoSchedule()`